db/snapshot-ctl.cc

```diff
@@ -144,82 +147,57 @@ future<> snapshot_ctl::clear_snapshot(sstring tag, std::vector<sstring> keyspace
 
 future<std::unordered_map<sstring, std::vector<snapshot_ctl::snapshot_details>>>
 snapshot_ctl::get_snapshot_details() {
-    using cf_snapshot_map = std::unordered_map<utils::UUID, column_family::snapshot_details>;
-    using snapshot_map = std::unordered_map<sstring, cf_snapshot_map>;
+    static auto details_result_hash = [] (const database::snapshot_details_result& sdr) {
+        return utils::tuple_hash()(std::make_tuple(sdr.snapshot_name, sdr.details.ks, sdr.details.cf));
+    };
+
+    // only one entry per {snapshot_name, ks, cf} triplet
```

Comment:
Could it happen that you get multiple entries with same snapshot name, ks and cf, but different `total`/`live`, because you're using default definitions of `operator==`? This doesn't directly follow from the definition of the map (because of `operator==`), but I guess it's simply impossible to have two such snapshots.

Reply:
Yes, this is my understanding of the original implementation. There should be no snapshots that differ only in `total`/`live` as they are merely copies of each other that we just have to reduce into one. That's what the original was doing, the only difference being that it retained the last one, while I keep the first one.